File: appmonet-sdk/appmonet-sdk-core/src/main/java/com/monet/bidder/Subscriber.java
package com.monet.bidder;

/**
 * This interface provides a blueprint for handling messages getting broadcasted.
 */
public interface Subscriber {

    /**
     * This method is called by {@link PubSubService} when a message is broadcasted.
     *
     * @param subscriberMessages The message getting broadcasted.
     */
    void onBroadcast(MonetPubSubMessage subscriberMessages);
}

File: appmonet-sdk/appmonet-sdk-core/src/main/java/com/monet/bidder/Subscriber.kt
package com.monet.bidder

/**
 * This interface provides a blueprint for handling messages getting broadcasted.
 */
interface Subscriber {
  /**
   * This method is called by [PubSubService] when a message is broadcasted.
   *
   * @param subscriberMessages The message getting broadcasted.
   */
  fun onBroadcast(subscriberMessages: MonetPubSubMessage?)
}